### ISSUE TITLE
Binary operator that returns `NoReturn` is evaluated as invalid

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5711]
```python
from typing import Any, NoReturn, overload


class Goo:
    @overload
    def __add__(self, other: str) -> NoReturn:
        ...

    @overload
    def __add__(self, other: Any) -> bool:
        ...

    def __add__(self, other: Any) -> bool:
        return True


y = Goo()
y + 4
y + "abc" # Error!
```

Pyright should not emit a type error here.

This is related to mypy issue https://github.com/python/mypy/issues/15861.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
